src/Engine/ActionExpressionContainer.php

```diff
@@ -162,4 +162,30 @@ public function checkIntegrity(ExecutionMetadataStateInterface $metadata_state)
     return $violation_list;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function prepareExecutionMetadataState(ExecutionMetadataStateInterface $metadata_state, ExpressionInterface $until = NULL) {
+    if ($until) {
+      if ($this->getUuid() === $until->getUuid()) {
+        return TRUE;
+      }
+      foreach ($this->actions as $action) {
+        if ($action->getUuid() === $until->getUuid()) {
```

Comment:
"===" is the sane default when comparing stuff that must be identical, so I would leave that as is.